sw/cheri/common/platform-pinmux.hh

```diff
@@ -0,0 +1,360 @@
+/**
+ * Copyright lowRISC contributors.
+ * Licensed under the Apache License, Version 2.0, see LICENSE for details.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+#include <debug.hh>
+#include <stdint.h>
+#include <utils.hh>
+
+/**
+ * A driver for Sonata's Pin Multiplexer (Pinmux).
+ *
+ * This driver can be used to select which block output is placed on a given
+ * output pin, and to select which input pin is provided to a given block
+ * input. Sonata's pinmux only allows certain selections per output pin / block
+ * input, which this driver describes.
+ *
+ * Documentation source can be found at:
+ * https://github.com/lowRISC/sonata-system/blob/6914b30a2769f881a53afb912032cd65a6d14307/doc/ip/pinmux.md
```

Comment:
I think better not to link to a specific commit here, could make this an in tree reference (Documentation source can be found in the repository at `doc/ip/pinmux.md`) and then leave the rendered doc link as is?

Reply:
I used a specific commit because that is what the existing drivers in `cheriot-rtos` do, to ensure that they point to the documentation at the time that the driver was written (which makes it easy to identify if a driver is bugged or has just fallen out of date). Because this driver is only intended to live in `sonata-system` temporarily whilst we change and iterate on the pinmux config (avoid breakages from PRs across multiple repos), it doesn't make sense to make an in-tree reference when this will eventually exist at `cheriot-rtos/sdk/include/platform/sunburst/platform-pinmux.hh`.